src/custom/components/AccountDetails/index.tsx

```diff
@@ -38,22 +38,56 @@ import {
   IconWrapper,
   renderTransactions
 } from './AccountDetailsMod'
+import { ConnectedWalletInfo, useWalletInfo } from 'hooks/useWalletInfo'
+import { MouseoverTooltip } from 'components/Tooltip/TooltipMod'
 
 type AbstractConnector = Pick<ReturnType<typeof useActiveWeb3React>, 'connector'>['connector']
 
-export function formatConnectorName(connector?: AbstractConnector) {
+function getWalletName(connector?: AbstractConnector): string {
   const { ethereum } = window
   const isMetaMask = !!(ethereum && ethereum.isMetaMask)
-  const name = Object.keys(SUPPORTED_WALLETS)
-    .filter(
-      k => SUPPORTED_WALLETS[k].connector === connector && (connector !== injected || isMetaMask === (k === 'METAMASK'))
-    )
-    .map(k => SUPPORTED_WALLETS[k].name)[0]
-  return <WalletName>Connected with {name}</WalletName>
+
+  const walletTuple = Object.entries(SUPPORTED_WALLETS).filter(
+    ([walletType, { connector: walletConnector }]) =>
+      walletConnector === connector && (connector !== injected || isMetaMask === (walletType === 'METAMASK'))
+  )
+  return walletTuple[0]?.[1]?.name || 'Unknown wallet'
```

Comment:
I get a bit lost. I understand the signature, but the implementation is a bit hard to follow. 
Maybe at least some comment. 

Also, I thought we were getting all the info from `useWalletInfo` are these logics something we want in the hook? (so components don't need to know about strange algorithms to get our best guess in the walletName?

Reply:
This is my attempt to make the original uniswap code more readable. See David's comment https://github.com/gnosis/gp-swap-ui/pull/633#discussion_r634122893
Failed miserably, apparently.